Source/CarthageKit/Xcode.swift

```diff
@@ -1181,6 +1181,67 @@ public func parseSecuritySigningIdentities(securityIdentities securityIdentities
 		.ignoreNil()
 }
 
+public func projectsInDirectory(directoryURL: NSURL, configuration: String, platforms: Set<Platform> = []) -> SignalProducer<(ProjectLocator, [String]), CarthageError> {
+	let directoryLength = directoryURL.pathComponents!.count
+	return locateProjectsInDirectory(directoryURL)
+		.filter { project in
+			let fileURL = project.fileURL
+			let fileComponentsLength = fileURL.pathComponents!.count
+			return fileComponentsLength - directoryLength < 3 // within 2 subdirectories. Because we actually have a test for this.
```

Comment:
I think ignoring projects found inside a `Carthage` directory at the top-level of the project directory seems like the right thing to do.